ui/dashboards/src/components/TimeRangeControls/TimeRangeControls.tsx

```diff
@@ -36,10 +37,14 @@ const DEFAULT_HEIGHT = '34px';
 interface TimeRangeControlsProps {
   // The controls look best at heights >= 28 pixels
   heightPx?: number;
+
+  // Whether to show the refresh button or not
+  showRefresh?: boolean;
 }
 
-export function TimeRangeControls({ heightPx }: TimeRangeControlsProps) {
+export function TimeRangeControls({ heightPx, showRefresh = true }: TimeRangeControlsProps) {
   const { timeRange, setTimeRange, refresh } = useTimeRange();
+  // TODO: Remove this since it couples to the dashboard context
```

Comment:
I see that we use this to add to add a timeOption if the dashboard spec has a pastDuration that is not available in our list of options. Ideally, we can remove this and find a better solution than plugging into the dashboard context.

Reply:
I remember adding this now, it is important though since I know Mohit wanted to trim the list of options and there is a high chance a duration isn’t in our default options. But agreed, this could be accounted for outside of TimeRangeControls and passed in